src/OrchardCore/OrchardCore.Abstractions/Shell/Removing/IShellRemovingManager.cs

```diff
@@ -0,0 +1,20 @@
+using System.Threading.Tasks;
+
+namespace OrchardCore.Environment.Shell.Removing;
+
+/// <summary>
+/// Host service managing the removing of a given tenant.
+/// </summary>
+public interface IShellRemovingManager
+{
+    /// <summary>
+    /// Removes the provided tenant.
+    /// </summary>
+    /// <param name="shellSettings">The tenant <see cref="ShellSettings"/>.</param>
+    /// <param name="localResourcesOnly">
+    /// Indicates that only local (non shared) resources should be removed.
```

Comment:
```suggestion
    /// Indicates that only local (not shared among server nodes in case of a multi-node hosting evironment) resources should be removed.
```